utils/update_build_version.py

```diff
@@ -55,21 +57,26 @@ def describe(dir):
     Runs 'git describe', or alternately 'git rev-parse HEAD', in dir.  If
     successful, returns the output; otherwise returns 'unknown hash, <date>'."""
     try:
-        return command_output(["git", "describe"], dir).rstrip()
+        return command_output(['git', 'describe'], dir).rstrip()
     except:
         try:
-            return command_output(["git", "rev-parse", "HEAD"], dir).rstrip()
+            return command_output(['git', 'rev-parse', 'HEAD'], dir).rstrip()
         except:
             return 'unknown hash, ' + datetime.date.today().isoformat()
 
+
 def main():
     if len(sys.argv) != 4:
-        print 'usage: {0} <shaderc_dir> <spirv-tools_dir> <glslang_dir>'.format(sys.argv[0])
+        print(
+            'usage: {0} <shaderc_dir> <spirv-tools_dir> <glslang_dir>'.format(sys.argv[0]))
         sys.exit(1)
 
-    new_content = ('"shaderc ' + describe(sys.argv[1]).replace('"', '\\"') + '\\n"\n' +
-                   '"spirv-tools ' + describe(sys.argv[2]).replace('"', '\\"') + '\\n"\n' +
-                   '"glslang ' + describe(sys.argv[3]).replace('"', '\\"') + '\\n"\n')
+    tags = [
+        describe(p).decode('ascii').replace('"', '\\"')
```

Comment:
In Python2, `str` is `bytes`, but not in Python3.

https://docs.python.org/3/howto/pyporting.html#text-versus-binary-data

Here, `Popen.communicate()` returns `bytes`, which in Python3 is binary data, cannot be directly `+`'d (or invoking `str`-related methods) to `str`, which is text data.